libs/utils/samplerum.js

```diff
@@ -0,0 +1,90 @@
+const sendPing = (checkpoint, data) => {
+  const { weight, id } = window.hlx.rum;
+  const body = JSON.stringify({
+    weight,
+    id,
+    referer: window.location.href,
+    generation: window.hlx.RUM_GENERATION,
+    checkpoint,
+    ...data,
+  });
+  const url = `https://rum.hlx.page/.rum/${weight}`;
+  navigator.sendBeacon(url, body);
+  console.debug(`ping:${checkpoint}`, data);
+};
+
+/* c8 ignore next 30 */
+const initSampleRUM = (rum, checkpoint, data) => {
+  window.hlx = window.hlx || {};
+
+  rum.defer = [];
+
+  rum.drain = ((dfnname, fn) => {
+    rum[dfnname] = fn;
+    rum.defer
+      .filter(({ fnname }) => dfnname === fnname)
+      .forEach(({ fnname, args }) => rum[fnname](...args));
+  });
+
+  rum.on = (chkpnt, fn) => {
+    rum.cases[chkpnt] = fn;
+  };
+
+  rum.cases = {
+    cwv: () => rum.cwv(data) || true,
+    lazy: () => {
+      // use classic script to avoid CORS issues
+      const script = document.createElement('script');
```

Comment:
I'd use our loadScript function.

Reply:
...but you may get a circular reference warning, which means you'd need to pass the function in when calling this.